### PR TITLE
chore: simplify lakefile

### PR DESCRIPTION
**Pull request by kim-em**
Per discussion with Mario and Mac last night.

### CODE REVIEW

**Comment by fgdorais** [comment]
Should we use TOML format?

**Comment by semorrison** [comment]
Yes, but I wanted to go in two steps. I'll do that as soon as this is merged.

**Comment by fgdorais** [comment]
Perhaps also remove `-Kwerror` from the build workflow.

**Comment by semorrison** [comment]
> Perhaps also remove `-Kwerror` from the build workflow.

Separate PR please? Seems orthogonal to this.

**Comment by fgdorais** [comment]
Does `-Kwerror` do anything without this lakefile? Sorry if I missed that somewhere upstream.

**Comment by digama0** [comment]
No, I think we still want `-Kwerror`. That option is the reason why we don't need `warningAsError=true` anymore (and conditional options in general are the main blocker for moving batteries' lakefile to toml format).

**Comment by fgdorais** [comment]
Ah! Found it. I think it's called `--wfail` though, I'm pretty sure `-Kwerror` won't do anything after changing the lakefile.

**Comment by kim-em** [comment]
@fgdorais, I can't tell from your last message what you want done here.

**Comment by fgdorais** [comment]
The `-Kwerror` in the build.yml file should be replaced by `--wfail`.

With this new lakefile `-Kwerror` does nothing: lake passes the `werror` option to the lakefile and the lakefile ignores it. To keep fail-on-warning working, the correct option is `--wfail` which tells lake itself to fail-on-warning. (I originally thought disabling fail-on-warning was part of the plan you came up with Mac and Mario, which is the root of all this confusion. I think we're all on the same page now.)

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
         name: build, test, and lint batteries
         uses: leanprover/lean-action@v1
         with:
-          build-args: '-Kwerror'
+          build-args: '---wfail'
```
Comment (fgdorais) [inline .github/workflows/build.yml]:
```suggestion
          build-args: '--wfail'
```

**Review by fgdorais — APPROVED** [review]

**Pull request merged**